pkg/jobWorker/job.go

```
@@ -0,0 +1,307 @@
+		Cloneflags: syscall.CLONE_NEWNS |
+			syscall.CLONE_NEWIPC |
+			syscall.CLONE_NEWNET |
+			syscall.CLONE_NEWUTS |
+			syscall.CLONE_NEWPID |
+			syscall.CLONE_NEWUSER,
+		UidMappings: []syscall.SysProcIDMap{
+			{
+				ContainerID: 0,
+				HostID:      os.Getuid(),
+				Size:        1,
+			},
+		},
+		GidMappings: []syscall.SysProcIDMap{
+			{
+				ContainerID: 0,
+				HostID:      os.Getgid(),
+				Size:        1,
+			},
+		},
+		// Also, enables mounting a new proc filesystem so that command such as `ps -ef` only see the processes in the PID namespace
+		Unshareflags: syscall.CLONE_NEWNS,
+		// instruct cmd.Run to use the control group file descriptor, so that Job Command does not
+		// have to manually add the new PID to the control group
+		UseCgroupFD: true,
+	}
+
+	formatedUUID := strings.Replace(job.UUID.String(), "-", "", -1)
+
+	cgroupName := formatedUUID
+	err := ns.CreateCGroup(cgroupName)
+	if err != nil {
+		return fmt.Errorf("error creating cgroup: %w", err)
+	}
+
+	if err = ns.AddResourceControl(cgroupName, ns.CPU_WEIGHT_File, strconv.Itoa(int(job.config.CPU*100))); err != nil {
+		log.Printf("could not add resources into controller:%s, %v", ns.CPU_WEIGHT_File, err)
+		return fmt.Errorf("error starting command: %w", err)
+	}
+	if err = ns.AddResourceControl(cgroupName, ns.MEMORY_HIGH_File, strconv.FormatInt(job.config.MemBytes, 10)); err != nil {
+		return fmt.Errorf("could not add resources into controller:%s, %v", ns.MEMORY_HIGH_File, err)
+	}
+	//if err = ns.AddResourceControl(cgroupName, ns.IO_WEIGHT_File, strconv.FormatInt(job.config.IOBytesPerSecond, 10)); err != nil {
+	//	return fmt.Errorf("could not add resources into controller:%s, %v", ns.IO_WEIGHT_File, err)
+	//}
+
+	//provide the file descriptor to cmd.Run so that it can add the new PID to the control group
+	if err = ns.AddProcess(cgroupName, cmd); err != nil {
+		log.Printf("Error AddProcess /proc - %s\n", err)
+		os.Exit(1)
```

This will result in the process using the library terminating.
Why not simply return an error and allow the caller to decide how to handle it?

---

@smallinsky agree. Make sense.